modules/osint/cve_search.py

```diff
@@ -0,0 +1,99 @@
+"""
+OWASP Maryam!
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+from concurrent.futures import thread
+from core.module import BaseModule
+import re
+import concurrent.futures
+
+class Module(BaseModule):
+
+    meta = {
+        'name': 'Common Vulnerabilities and Exposures Searcher',
+        'author': 'Dimitrios Papageorgiou',
+        'version': '0.1',
+        'description': 'Search in open-sources to find CVEs.',
+        'sources': ('mitre'),
+        'options': (
+                ('query', None,
+                 True, 'Query string(bug name, app name, etc)', '-q', 'store'),
+                ('sources', 'mitre', False,
+                 'DB source to search. e.g mitre,...(default=mitre)', '-s', 'store'),
+                ('thread', 2, False,
+                 'The number of engine that run per round(default=2)', '-t', 'store'),
+                ('output', False, False, 'Save output to  workspace',
+                 '--output', 'store_true'),
+        ),
+        'examples': ('cve_search -q sql --output')
+    }
+
+    names = []
+    links = []
+    descriptions = []
+
+    def clear(self):
+        self.names = []
+        self.links = []
+        self.descriptions = []
+
+    def mitre(self, q):
+        self.verbose('[MITRE] Searching in mitre...')
+        try:
+            req = self.request(
+                f"https://cve.mitre.org/cgi-bin/cvekey.cgi?keyword={q}")
+        except Exception as e:
+            self.error('Mitre is missed!')
+        else:
+            self.names.extend(re.findall(
+                r'<a href="[^"]+">(CVE-[^<]+)</a>', req.text))
```

Comment:
*. is a dangerous pattern. Because sometimes match two or three tags.
e.g: 
`<a href="aaaa">bbb</a>
<a href="bbb">cccc</a>
`
matched pattern:
`aaaa">bbb</a>
<a href="bbb"`
So I changed it. Have some tests to be sure that it is correct.

Reply:
Ok cool thanks.